postgres/literal.go

```diff
@@ -138,3 +138,23 @@ func Timestampz(year int, month time.Month, day, hour, minute, second int, milli
 func TimestampzT(t time.Time) TimestampzExpression {
 	return CAST(jet.TimestampzT(t)).AS_TIMESTAMPZ()
 }
+
+// DateRange creates new date range literal expression from two dates
+func DateRange(lowDate, highDate time.Time, bounds string) RangeExpression {
+	return CAST(jet.DateRange(lowDate, highDate, bounds)).AS_DATERANGE()
```

Comment:
If `DateRange` is function constructor then additional cast is probably unnecessary. Because generated sql would be `daterange(start,end)::daterange`